client/landing/stepper/declarative-flow/internals/state-manager/store.ts

```diff
@@ -31,6 +31,9 @@ export function useFlowState() {
 
 	const { data: state } = useQuery< FlowStateManifest >( {
 		queryKey: [ PREFIX, flow, session, VERSION ],
+		queryFn: ( { client, queryKey } ) => {
+			return ( client.getQueryData( queryKey ) ?? {} ) as FlowStateManifest;
+		},
```

Comment:
I'm a bit worried about this being cyclical. Because React Query will think this data is fresh since it came from the query function and we don't know what assumptions we will break with this.

Maybe we should get rid of `useQuery` and inside `get` function we simply do 
```js
const key = [ PREFIX, flow, session, VERSION ];
const data = client.getQueryData( key ) ?? {} ) as FlowStateManifest;
```

WDYT?

Reply:
Actually, it doesn't matter because `setQueryData` triggers a re-render so the user will get the up to date information. Pushed a patch!

Reply:
`useQueryClient` is reactive.